test/components/birth-date-picker.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import { ClientRenderer, expect, simulate, sinon, waitFor } from "test-drive-react";
+import { BirthDatePicker, dateFromYearMonthDay } from "../../src";
+import { BirthDatePickerDemo } from "../../demo/birth-date-picker-demo";
+
+function updateInput(input: HTMLInputElement, value: string) {
+    input.focus();
+    input.value = value;
+    simulate.change(input);
+}
+
+describe("<BirthDatePicker />", () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    it("Displays the provided date and allows changing it", async function() {
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePickerDemo />);
+
+        await waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER")).to.be.present();
+        });
+
+        const yearInput = select("BIRTH_DATE_PICKER_YEAR") as HTMLInputElement;
+        const monthInput = select("BIRTH_DATE_PICKER_MONTH") as HTMLInputElement;
+        const dayInput = select("BIRTH_DATE_PICKER_DAY") as HTMLInputElement;
+
+        expect(yearInput).to.have.value("2001");
+        expect(monthInput).to.have.value("9");
+        expect(dayInput).to.have.value("11");
+
+        updateInput(yearInput, "2002");
+        updateInput(monthInput, "10");
+        updateInput(dayInput, "12");
+
+        return waitFor(() => {
+            expect(yearInput).to.have.value("2002");
+            expect(monthInput).to.have.value("10");
+            expect(dayInput).to.have.value("12");
+        });
+    });
+
+    it("Renders with default props", function () {
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePicker />);
+
+        return waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER")).to.be.present();
+            expect(select("BIRTH_DATE_PICKER_YEAR")).to.have.value("");
+            expect(select("BIRTH_DATE_PICKER_MONTH")).to.have.value("");
+            expect(select("BIRTH_DATE_PICKER_DAY")).to.have.value("");
+        });
+    });
+
+    it("Displays the provided date", function() {
+        const value = new Date("1987-04-26Z");
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePicker value={value} />);
+
+        return waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER_YEAR")).to.have.value("1987");
+            expect(select("BIRTH_DATE_PICKER_MONTH")).to.have.value("4");
+            expect(select("BIRTH_DATE_PICKER_DAY")).to.have.value("26");
+        });
+    });
+
+    it("Emits onChange when one of the inputs is changed", async function() {
```

Comment:
I suggest splitting this ("aggregated") test into several. It will be much easier to follow.